### PR TITLE
React Router v6 + Auth Flow

### PR DESCRIPTION
**Pull request by dylancom**
- Upgraded from v5. (Many API changes)
- Added auth-flow (using context). Protected screens will not be viewable anymore when locked.

### CODE REVIEW

**Comment by bumi** [comment]
No matching routes for adding a new account from the options page: 

```
-  No routes matched location "/accounts/new/lnd-hub" index.js:21:48
- No routes matched location "/accounts/new/lnd" index.js:21:48
```

**Comment by dylancom** [comment]
Fixed

File: src/app/context/AuthContext.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+const AuthContext = createContext<AuthContextType>(null!);
+
+export function AuthProvider({ children }: { children: React.ReactNode }) {
+  const [user, setUser] = useState<any>(null);
+  const [loading, setLoading] = useState(true);
+
+  const unlock = (password: string, callback: VoidFunction) => {
+    return utils.call("unlock", { password }).then(() => {
+      setUser("alby");
```
[the remark this answers is not in the document]
Comment (bumi) [reply src/app/context/AuthContext.tsx:19]: I am wondering if calling it `user` is the best here... we will always only have one user, but multiple accounts. maybe we can "authenticate" towards one account? and store here which account is currently used?

**Review by bumi — APPROVED** [review]
I can't say much about the code and it is hard to compare. So far it works great, I will keep testing it more and we should soon merge this to prevent further merge conflicts. 
can you maybe rebase this one?

File: src/app/context/AuthContext.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (dylancom) [reply src/app/context/AuthContext.tsx:19]: Currently "alby" is just a placeholder, could be "user" aswel. We only check for the presence of a value inside the app. But we could store here also which account is currently used.

[the remark this answers is not in the document]
Comment (bumi) [reply src/app/context/AuthContext.tsx:19]: if we call it account or something like currentAccount then we would not introduce another terminology/entity.

File: src/app/context/AuthContext.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          utils.openPage("welcome.html");
+          window.close();
+        } else if (response.unlocked) {
+          setAccount(response.currentAccountId);
+        } else {
+          setAccount(null);
+        }
+      })
+      .catch((e) => {
+        console.log(e);
```
Comment (bumi) [inline src/app/context/AuthContext.tsx:45]:
```suggestion
        console.log(e);
        alert(`An unexpected error occurred (${e.message})`);
```

File: src/app/screens/Publisher.tsx [hunk 3]
```diff
@@ -24,16 +24,18 @@ function Publisher() {
...
-      const response = await utils.call("getAllowanceById", {
-        id: parseInt(id),
-      });
-      console.log(response);
-      setAllowance(response);
+      if (id) {
+        const response = await utils.call("getAllowanceById", {
+          id: parseInt(id),
+        });
+        console.log(response);
```
Comment (bumi) [inline src/app/screens/Publisher.tsx:36]:
```suggestion
```

File: src/extension/background-script/actions/accounts/unlock.js [hunk 4]
```diff
@@ -18,7 +19,10 @@ const unlock = (message, sender) => {
 
   // if everything is fine we keep the password in memory
   state.setState({ password });
-  return Promise.resolve({ data: { unlocked: true } });
+
+  console.log("account", account);
```
Comment (bumi) [inline src/extension/background-script/actions/accounts/unlock.js:23]:
```suggestion
```

**Review by bumi — APPROVED** [review]
had some minor comments about some debug logs that we can remove probably?

Let's merge it and then test the app in detail again. @MoritzKa can you also do another testing round

**Comment by bumi** [comment]
@dylancom can you rebase and then merge?

**Pull request merged**